prboom2/src/g_game.c

```diff
@@ -3512,21 +3578,175 @@ const byte* G_ReadDemoHeaderEx(const byte *demo_p, size_t size, unsigned int par
+      I_Error("G_ReadDemoHeader: Extended demo format version %d unrecognized.", extension_version);
+    }
+
+    num_extensions  =                 *demo_p++;
+    num_extensions |= ((unsigned int)(*demo_p++)) <<  8;
+
+    if (CheckForOverrun(header_p, demo_p, size, num_extensions, failonerror))
+      return NULL;
+
+    for (i = 0; i < num_extensions; i++)
+    {
+      int r_len = *demo_p++;
+
+      if (CheckForOverrun(header_p, demo_p, size, r_len, failonerror))
+        return NULL;
+
+      // ano - jun2019 - when more potential extension strings get added,
+      // this section can become more complex
+      if (r_len == 8 && strncmp((const char *)demo_p, "UMAPINFO", 8) == 0)
+      {
+        using_umapinfo = 1;
+      }
+      else
+      {
+        // ano - TODO better error handling here?
+        I_Error("G_ReadDemoHeader: Extended demo format extension unrecognized.");
+      }
+
+      demo_p += r_len;
+    }
+
+    // ano - jun2019 - load lump name if we're using umapinfo
+    // this is a bit hacky to just read in episode / map number from string
+    // currently PR+ doesn't support arbitrary mapnames, but one day it may,
+    // so this is for forward compat
+    if(using_umapinfo)
+    {
+      const byte *string_end = demo_p + 8;
+
+      if (CheckForOverrun(header_p, demo_p, size, 8, failonerror))
+        return NULL;
+
+      if (strncmp((const char *)demo_p, "MAP", 3) == 0)
+      {
+        // MAPx form
+        episode = 1;
+        demo_p += 3;
+        map = 0;
+
+        // we're doing hellworld number parsing and i'm sorry
+        // CAPTAIN, WE'RE PICKING UP SOME CODE DUPLICATION IN SECTOR 47
+        for (i = 0; demo_p < string_end; i++)
+        {
+          char cur = *demo_p++;
+          
+          if (cur < '0' || cur > '9')
+          {
+            if (cur != 0)
+            {
+              I_Error("G_ReadDemoHeader: Unable to determine map for UMAPINFO demo.");
+            }
+            break;
+          }
+
+          map = (map * 10) + (cur - '0');
+        }
+      }
+      else if(*demo_p++ == 'E')
+      {
+        // EyMx form
+        episode = 0;
+        map = 0;
+
+        // read in episode #
+        for (i = 0; demo_p < string_end; i++)
+        {
+          char cur = *demo_p++;
+
+          if (cur < '0' || cur > '9')
+          {
+            if (cur == 0 || cur == 'M')
+            {
+              break;
+            }
+
+            I_Error("G_ReadDemoHeader: Unable to determine map for UMAPINFO demo.");
+          }
+
+          episode = (episode * 10) + (cur - '0');
+        }
+
+        // read in map #
+        for (i = 0; demo_p < string_end; i++)
+        {
+          char cur = *demo_p++;
+
+          if (cur < '0' || cur > '9')
+          {
+            if (cur == 0)
+            {
+              break;
+            }
+
+            I_Error("G_ReadDemoHeader: Unable to determine map for UMAPINFO demo.");
+          }
+
+          episode = (episode * 10) + (cur - '0');
```

Comment:
This should be `map` here instead of `episode`, right?

Reply:
Yup. Fixed.